File: 1624. Largest Substring Between Two Equal Characters.java
class Solution {
    public int maxLengthBetweenEqualCharacters(String s) {
        int len = -1;
        for (int i = 0; i < s.length(); i++) {
            for (int j = s.length()-1; j > i; j--) {
                if ((s.charAt(i) == s.charAt(j)) && ((j-i) > len)) {
                    len = (j-i) - 1;
                }
            }
        }
        return len;
    }
}

File: 1624. Largest Substring Between Two Equal Characters.cpp
class Solution {
public:
    int maxLengthBetweenEqualCharacters(string s) {
        int count = -1;
        int n = s.length();

        if (n == 2) {
            return 0;
        }
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if (s[i] == s[j]) {
                    count = max(count, j - i - 1);
                }
            }
        }
        return count;
    }
};
